Compilation with amdflang fails

On a system with ROCm 6.0.1 installed, attempts to install with `spack install feq-parse % rocmcc` fails with the following error

```
3 errors found in build log:
     49    [  2%] Building Fortran object src/CMakeFiles/feqparse.dir/FEQParse.F90.o
     50    cd /tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-build-mj5pd3f/src && /opt/spack/lib/spack/env/rocmcc/amdflang -Dfeqparse_EXPORTS  -O3 
           -DNDEBUG -J../include -fPIC -c /tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-src/src/FEQParse.F90 -o CMakeFiles/feqparse.dir/FEQParse.F
           90.o
     51    F90-S-0026-Unmatched quote (/tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-src/src/FEQParse.F90: 271)
     52      0 inform,   0 warnings,   1 severes, 0 fatal for tokenize
     53    F90-S-0026-Unmatched quote (/tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-src/src/FEQParse.F90: 1642)
     54      0 inform,   0 warnings,   1 severes, 0 fatal for priority
  >> 55    make[2]: *** [src/CMakeFiles/feqparse.dir/build.make:78: src/CMakeFiles/feqparse.dir/FEQParse.F90.o] Error 1
     56    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-build-mj5pd3f'
  >> 57    make[1]: *** [CMakeFiles/Makefile2:123: src/CMakeFiles/feqparse.dir/all] Error 2
     58    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-feq-parse-2.1.0-mj5pd3fuk5pau3org24ulsw7tedhwmo6/spack-build-mj5pd3f'
  >> 59    make: *** [Makefile:139: all] Error 2
```

Testing with v2.2.0 of feq-parse fails installation with rocmcc with the following error
```
3 errors found in build log:
     47    cd /tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-build-3fpb6fg/src && /opt/spack/lib/spack/env/rocmcc/amdflang -Dfeqparse_EXPORTS  -O3 
           -DNDEBUG -J../include -fPIC -c /tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-src/src/FEQParse_TokenStack.F90 -o CMakeFiles/feqparse.dir
           /FEQParse_TokenStack.F90.o
     48    /usr/bin/cmake -E cmake_copy_f90_mod include/feqparse_tokenstack.mod src/CMakeFiles/feqparse.dir/feqparse_tokenstack.mod.stamp Flang
     49    [  1%] Building Fortran object src/CMakeFiles/feqparse.dir/FEQParse.F90.o
     50    cd /tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-build-3fpb6fg/src && /opt/spack/lib/spack/env/rocmcc/amdflang -Dfeqparse_EXPORTS  -O3 
           -DNDEBUG -J../include -fPIC -c /tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-src/src/FEQParse.F90 -o CMakeFiles/feqparse.dir/FEQParse.F
           90.o
     51    F90-F-0155-Derived type has not been declared - indepvar (/tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-src/src/FEQParse.F90: 57)
     52    F90/x86-64 Linux Flang - 1.5 2017-05-01: compilation aborted
  >> 53    make[2]: *** [src/CMakeFiles/feqparse.dir/build.make:78: src/CMakeFiles/feqparse.dir/FEQParse.F90.o] Error 1
     54    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-build-3fpb6fg'
  >> 55    make[1]: *** [CMakeFiles/Makefile2:461: src/CMakeFiles/feqparse.dir/all] Error 2
     56    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-feq-parse-2.2.0-3fpb6fg7jvhuzuccwyzti6zuai2z4f7f/spack-build-3fpb6fg'
  >> 57    make: *** [Makefile:139: all] Error 2
```

--- a/.github/workflows/linux-amdflang-cmake.yaml
+++ b/.github/workflows/linux-amdflang-cmake.yaml
@@ -17,7 +17,7 @@ on:
 
 jobs:
   linux-tests:
-    timeout-minutes: 8
+    timeout-minutes: 20
     if: "!contains(github.event.head_commit.message, 'skip ci')"
     name: ${{ matrix.os }} - ${{ matrix.fcompiler }} - ${{ matrix.build_type }}
     runs-on: ${{ matrix.os }}
@@ -25,7 +25,7 @@ jobs:
       fail-fast: false
       matrix:
         include:
-            - os: ubuntu-latest-m
+            - os: ubuntu-22.04
               fcompiler: amdflang
               ccompiler: amdclang
               shell: bash

--- a/.github/workflows/linux-amdflang-cmake.yaml
+++ b/.github/workflows/linux-amdflang-cmake.yaml
@@ -45,7 +45,7 @@ jobs:
            sudo apt update -y
            wget https://repo.radeon.com/amdgpu-install/6.1.1/ubuntu/jammy/amdgpu-install_6.1.60101-1_all.deb
            sudo apt install -y ./amdgpu-install_6.1.60101-1_all.deb
-           sudo amdgpu-install --accept-eula -y --usecase=rocm --no-dkms
+           sudo amdgpu-install --accept-eula -y --usecase=openmpsdk --no-dkms
 
       - name: Show version information
         run: |

--- a/.github/workflows/linux-amdflang-cmake.yaml
+++ b/.github/workflows/linux-amdflang-cmake.yaml
@@ -1,0 +1,68 @@
+name: linux-amdflang-cmake
+
+on:
+  push:
+    branches:
+      - master
+      - main
+    paths-ignore:
+      - 'AUTHORS.md'
+      - 'LICENSE.md'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - 'AUTHORS.md'
+      - 'LICENSE.md'
+      - 'README.md'
+
+jobs:
+  linux-tests:
+    timeout-minutes: 8
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    name: ${{ matrix.os }} - ${{ matrix.fcompiler }} - ${{ matrix.build_type }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+            - os: ubuntu-latest-m
+              fcompiler: amdflang
+              ccompiler: amdclang
+              shell: bash
+              build_type: debug
+              memcheck: false
+
+    defaults:
+      run:
+        shell: ${{ matrix.shell }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+
+      - name: rocm-setup
+        run: |
+           sudo apt update -y
+           wget https://repo.radeon.com/amdgpu-install/6.1.1/ubuntu/jammy/amdgpu-install_6.1.60101-1_all.deb
+           sudo apt install -y ./amdgpu-install_6.1.60101-1_all.deb
+           sudo amdgpu-install --accept-eula -y --usecase=rocm --no-dkms
+
+      - name: Show version information
+        run: |
+          export PATH=${PATH}:/opt/rocm/bin
+          ${{ matrix.fcompiler }} --version
+          ${{ matrix.ccompiler }} --version
+
+      - name: Build with Cmake
+        run: |
+          export PATH=${PATH}:/opt/rocm/bin
+          mkdir build
+          cd build
+          FC=${{ matrix.fcompiler }} CC=${{ matrix.ccompiler }} cmake -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}  ../
+          make VERBOSE=1
+
+      - name: Run ctests
+        run: |
+          cd build/test
+          ctest || ctest --rerun-failed --output-on-failure
+

--- a/.github/workflows/linux-nvfortran-cmake.yaml
+++ b/.github/workflows/linux-nvfortran-cmake.yaml
@@ -1,0 +1,65 @@
+name: linux-amdflang-cmake
+
+on:
+  push:
+    branches:
+      - master
+      - main
+    paths-ignore:
+      - 'AUTHORS.md'
+      - 'LICENSE.md'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - 'AUTHORS.md'
+      - 'LICENSE.md'
+      - 'README.md'
+
+jobs:
+  linux-tests:
+    timeout-minutes: 20
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    name: ${{ matrix.os }} - ${{ matrix.toolchain.compiler }} - ${{ matrix.build_type }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - {compiler: nvidia-hpc, version: '23.11'}
+        include:
+            - os: ubuntu-22.04
+              shell: bash
+              build_type: debug
+              toolchain: {compiler: nvidia-hpc, version: '23.11'}
+
+    defaults:
+      run:
+        shell: ${{ matrix.shell }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - uses: fortran-lang/setup-fortran@v1
+        id: setup-fortran
+        with:
+          compiler: ${{ matrix.toolchain.compiler }}
+          version: ${{ matrix.toolchain.version }}
+
+      - name: Show version information
+        run: |
+          ${{ env.FC }} --version
+          ${{ env.CC }} --version
+
+      - name: Build with Cmake
+        run: |
+          export PATH=${PATH}:/opt/rocm/bin
+          mkdir build
+          cd build
+          FC=${{ env.FC }} CC=${{ env.CC }} cmake -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}  ../
+          make VERBOSE=1
+
+      - name: Run ctests
+        run: |
+          cd build/test
+          ctest || ctest --rerun-failed --output-on-failure
+

--- a/.github/workflows/linux-nvfortran-cmake.yaml
+++ b/.github/workflows/linux-nvfortran-cmake.yaml
@@ -1,4 +1,4 @@
-name: linux-amdflang-cmake
+name: linux-nvidia-hpc-cmake
 
 on:
   push:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,13 +45,20 @@ elseif( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "IntelLLVM" )
     set( CMAKE_C_FLAGS_COVERAGE "-g -O0")
 
 elseif( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Flang" )
-    set( CMAKE_Fortran_FLAGS "-Mpreprocess -fstack-arrays" )
-    set( CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS} -O0 -g -Wall -Mstandard -pedantic" )
+    set( CMAKE_Fortran_FLAGS "-Mpreprocess" )
+    set( CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS} -O0 -g -Wall -pedantic" )
     set( CMAKE_Fortran_FLAGS_COVERAGE "${CMAKE_Fortran_FLAGS_DEBUG}") # Code coverage not available with ifx
     set( CMAKE_Fortran_FLAGS_PROFILE "-O3")
     set( CMAKE_Fortran_FLAGS_RELEASE "-O3" )
     set( CMAKE_C_FLAGS_COVERAGE "-g -O0")
 
+    elseif( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "NVHPC" )
+    set( CMAKE_Fortran_FLAGS "-Mpreprocess" )
+    set( CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS} -O0 -Wall -C -g -traceback" )
+    set( CMAKE_Fortran_FLAGS_COVERAGE "${CMAKE_Fortran_FLAGS_DEBUG}") # Code coverage not available with ifx
+    set( CMAKE_Fortran_FLAGS_PROFILE "-O3 -Mstack_arrays -Minfo")
+    set( CMAKE_Fortran_FLAGS_RELEASE "-O3 -Mstack_arrays -Minfo" )
+    set( CMAKE_C_FLAGS_COVERAGE "-g -O0")
 endif()
 
 ADD_SUBDIRECTORY(${CMAKE_SOURCE_DIR}/src)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,8 @@ ENDIF(NOT CMAKE_Fortran_COMPILER_SUPPORTS_F90)
 # Default Fortran compiler flags
 # Fortran compiler flags
 
+message(CMAKE_Fortran_COMPILER_ID="${CMAKE_Fortran_COMPILER_ID}")
+
 if( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU" )
     set( CMAKE_Fortran_FLAGS "-cpp -ffree-line-length-512" )
     set( CMAKE_Fortran_FLAGS_DEBUG "-g -O0 -C -Wall -fbounds-check -fbacktrace -ffpe-trap=invalid,zero,overflow" )
@@ -41,6 +43,15 @@ elseif( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "IntelLLVM" )
     set( CMAKE_Fortran_FLAGS_PROFILE "-O3")
     set( CMAKE_Fortran_FLAGS_RELEASE "-O3" )
     set( CMAKE_C_FLAGS_COVERAGE "-g -O0")
+
+elseif( "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Flang" )
+    set( CMAKE_Fortran_FLAGS "-Mpreprocess -fstack-arrays" )
+    set( CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS} -O0 -g -Wall -Mstandard -pedantic" )
+    set( CMAKE_Fortran_FLAGS_COVERAGE "${CMAKE_Fortran_FLAGS_DEBUG}") # Code coverage not available with ifx
+    set( CMAKE_Fortran_FLAGS_PROFILE "-O3")
+    set( CMAKE_Fortran_FLAGS_RELEASE "-O3" )
+    set( CMAKE_C_FLAGS_COVERAGE "-g -O0")
+
 endif()
 
 ADD_SUBDIRECTORY(${CMAKE_SOURCE_DIR}/src)

--- a/README.md
+++ b/README.md
@@ -1,16 +1,18 @@
 # Fortran Equation Parser
 Copyright 2020 Fluid Numerics LLC
 
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-gnu-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-gnu-cmake.yml)
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-gnu-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-gnu-fpm.yml)
+[![linux-gnu-cmake](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-gnu-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-gnu-cmake.yml)
+[![linux-gnu-fpm](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-gnu-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-gnu-fpm.yml)
 
 
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-intel-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-intel-cmake.yml)
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-intel-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-intel-fpm.yml)
+[![linux-intel-cmake](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-intel-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-intel-cmake.yml)
+[![linux-intel-fpm](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-intel-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-intel-fpm.yml)
+
+[![linux-amdflang-cmake](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-amdflang-cmake.yaml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-amdflang-cmake.yaml)
 
 
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml)
-[![Build Status](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml)
+[![windows-gnu-cmake](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml)
+[![windows-gnu-fpm](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml)
 
 [![codecov](https://codecov.io/gh/FluidNumerics/feq-parse/graph/badge.svg?token=IBNDDI4MHB)](https://codecov.io/gh/FluidNumerics/feq-parse)
 

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ Copyright 2020 Fluid Numerics LLC
 [![linux-intel-fpm](https://github.com/fluidnumerics/feq-parse/actions/workflows/linux-intel-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-intel-fpm.yml)
 
 [![linux-amdflang-cmake](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-amdflang-cmake.yaml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-amdflang-cmake.yaml)
-
+[![linux-nvfortran-cmake](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-nvfortran-cmake.yaml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/linux-nvfortran-cmake.yaml)
 
 [![windows-gnu-cmake](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-cmake.yml)
 [![windows-gnu-fpm](https://github.com/fluidnumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml/badge.svg)](https://github.com/FluidNumerics/feq-parse/actions/workflows/windows-gnu-fpm.yml)

--- a/src/FEQParse.F90
+++ b/src/FEQParse.F90
@@ -330,16 +330,13 @@ contains
     call operator_stack%Construct(Stack_Length)
 
     do i = 1,parser%infix%top_index
-      print*, parser % inFix % tokens(i) % tokenString
       if(parser%inFix%tokens(i)%tokenType == Variable_Token .or. &
          parser%inFix%tokens(i)%tokenType == Number_Token) then
 
-        print*, 'convert : parser : push variable/number'
         call parser%postFix%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == Function_Token) then
 
-        print*, 'convert : operator : push function'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == Operator_Token &
@@ -354,10 +351,7 @@ contains
                    parser%Priority(parser%inFix%tokens(i)) .and. &
                    .not. operator_stack%IsEmpty())
 
-            
-            print*, 'convert : parser :  push operator/monad'
             call parser%postFix%push(tok)
-            print*, 'convert : operator :  pop operator/monad'
             call operator_stack%pop(tok)
             tok = operator_stack%TopToken()
 
@@ -365,12 +359,10 @@ contains
 
         endif
 
-        print*, 'convert : operator :  push operator/monad'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == OpeningParentheses_Token) then
 
-        print*, 'convert : operator :  push open par'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == ClosingParentheses_Token) then
@@ -379,30 +371,24 @@ contains
 
         do while(.not.(operator_stack%IsEmpty()) .and. tok%tokenString(1:1) /= '(')
 
-          print*, 'convert : parser :  push '//trim(tok%tokenString)
           call parser%postFix%push(tok)
-          print*, 'convert : operator :  pop '//trim(tok%tokenString)
           call operator_stack%pop(tok)
           tok = operator_stack%TopToken()
 
         enddo
 
         ! Pop the opening parenthesis
-        print*, 'convert : operator :  pop '//trim(tok%tokenString)
         call operator_stack%pop(tok)
 
       endif
 
     enddo
 
-    print*,'----- pop remaining operator ----- '
     ! Pop the remaining operators
     do while(.not.(operator_stack%IsEmpty()))
 
       tok = operator_stack%TopToken()
-      print*, 'convert : parser :  push '//trim(tok%tokenString)
       call parser%postFix%push(tok)
-      print*, 'convert : operator :  pop '//trim(tok%tokenString)
       call operator_stack%pop(tok)
 
     enddo
@@ -850,7 +836,7 @@ contains
     do k = 1,parser%postfix%top_index
 
       t = parser%postfix%tokens(k)%Copy()
-      print*, TRIM(t%tokenString)
+      print*,trim(t%tokenString)
 
       select case(t%tokenType)
 
@@ -862,14 +848,12 @@ contains
           v = vnumber
         endif
 
-        print*,'push'
         call stack%Push(v)
 
       case(Variable_Token)
 
         do i = 1,parser%nIndepVars
           if(trim(t%tokenString) == parser%indepVars(i)%value) then
-            print*,'push'
             call stack%Push(x(:,:,i))
             exit
           endif
@@ -877,9 +861,7 @@ contains
 
       case(Operator_Token)
 
-        print*,'pop'
         call stack%Pop(a)
-        print*,'pop'
         call stack%Pop(b)
 
         select case(trim(t%tokenString))
@@ -907,27 +889,22 @@ contains
 
         endselect
 
-        print*,'push'
         call stack%Push(c)
 
       case(Function_Token)
 
-        print*,'pop'
         call stack%Pop(a)
 
         b = Functions(t%tokenIndex)%invoke(a)
 
-        print*,'push'
         call stack%Push(b)
 
       case(Monadic_Token)
 
         if(trim(t%tokenString) == '-') then
 
-          print*,'pop'
           call stack%Pop(a)
           a = -a
-          print*,'push'
           call stack%Push(a)
 
         endif

--- a/src/FEQParse.F90
+++ b/src/FEQParse.F90
@@ -49,6 +49,10 @@ module FEQParse
 
   private
 
+  type IndepVar
+    character(:),allocatable :: value
+  endtype
+
   type,public :: EquationParser
     character(:),allocatable                   :: equation
     character(:),allocatable                   :: variableName
@@ -80,10 +84,6 @@ module FEQParse
     procedure,private :: Priority
 
   endtype EquationParser
-
-  type IndepVar
-    character(:),allocatable :: value
-  endtype
 
   interface EquationParser
     procedure Construct_EquationParser
@@ -330,14 +330,16 @@ contains
     call operator_stack%Construct(Stack_Length)
 
     do i = 1,parser%infix%top_index
-      !print*, parser % inFix % tokens(i) % tokenString
+      print*, parser % inFix % tokens(i) % tokenString
       if(parser%inFix%tokens(i)%tokenType == Variable_Token .or. &
          parser%inFix%tokens(i)%tokenType == Number_Token) then
 
+        print*, 'convert : parser : push variable/number'
         call parser%postFix%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == Function_Token) then
 
+        print*, 'convert : operator : push function'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == Operator_Token &
@@ -349,9 +351,13 @@ contains
 
           do while(trim(tok%tokenString) /= '(' .and. &
                    parser%Priority(tok) > &
-                   parser%Priority(parser%inFix%tokens(i)))
+                   parser%Priority(parser%inFix%tokens(i)) .and. &
+                   .not. operator_stack%IsEmpty())
 
+            
+            print*, 'convert : parser :  push operator/monad'
             call parser%postFix%push(tok)
+            print*, 'convert : operator :  pop operator/monad'
             call operator_stack%pop(tok)
             tok = operator_stack%TopToken()
 
@@ -359,10 +365,12 @@ contains
 
         endif
 
+        print*, 'convert : operator :  push operator/monad'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == OpeningParentheses_Token) then
 
+        print*, 'convert : operator :  push open par'
         call operator_stack%push(parser%inFix%tokens(i))
 
       elseif(parser%inFix%tokens(i)%tokenType == ClosingParentheses_Token) then
@@ -371,24 +379,30 @@ contains
 
         do while(.not.(operator_stack%IsEmpty()) .and. tok%tokenString(1:1) /= '(')
 
+          print*, 'convert : parser :  push '//trim(tok%tokenString)
           call parser%postFix%push(tok)
+          print*, 'convert : operator :  pop '//trim(tok%tokenString)
           call operator_stack%pop(tok)
           tok = operator_stack%TopToken()
 
         enddo
 
         ! Pop the opening parenthesis
+        print*, 'convert : operator :  pop '//trim(tok%tokenString)
         call operator_stack%pop(tok)
 
       endif
 
     enddo
 
+    print*,'----- pop remaining operator ----- '
     ! Pop the remaining operators
     do while(.not.(operator_stack%IsEmpty()))
 
       tok = operator_stack%TopToken()
+      print*, 'convert : parser :  push '//trim(tok%tokenString)
       call parser%postFix%push(tok)
+      print*, 'convert : operator :  pop '//trim(tok%tokenString)
       call operator_stack%pop(tok)
 
     enddo
@@ -836,6 +850,7 @@ contains
     do k = 1,parser%postfix%top_index
 
       t = parser%postfix%tokens(k)%Copy()
+      print*, TRIM(t%tokenString)
 
       select case(t%tokenType)
 
@@ -847,12 +862,14 @@ contains
           v = vnumber
         endif
 
+        print*,'push'
         call stack%Push(v)
 
       case(Variable_Token)
 
         do i = 1,parser%nIndepVars
           if(trim(t%tokenString) == parser%indepVars(i)%value) then
+            print*,'push'
             call stack%Push(x(:,:,i))
             exit
           endif
@@ -860,7 +877,9 @@ contains
 
       case(Operator_Token)
 
+        print*,'pop'
         call stack%Pop(a)
+        print*,'pop'
         call stack%Pop(b)
 
         select case(trim(t%tokenString))
@@ -888,22 +907,27 @@ contains
 
         endselect
 
+        print*,'push'
         call stack%Push(c)
 
       case(Function_Token)
 
+        print*,'pop'
         call stack%Pop(a)
 
         b = Functions(t%tokenIndex)%invoke(a)
 
+        print*,'push'
         call stack%Push(b)
 
       case(Monadic_Token)
 
         if(trim(t%tokenString) == '-') then
 
+          print*,'pop'
           call stack%Pop(a)
           a = -a
+          print*,'push'
           call stack%Push(a)
 
         endif

